vllm-model-provider/server/server.go

```diff
@@ -0,0 +1,125 @@
+package server
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"path"
+
+	openai "github.com/gptscript-ai/chat-completion-client"
+)
+
+func Run(apiKey, endpointStr, port string) error {
+	if endpointStr == "" {
+		return fmt.Errorf("endpoint is required")
+	}
+
```

Comment:
nit: already checked this, so can probably remove from here. Plus, it will be caugh below when you try to parse the URL.